packages/cardano-services/src/StakePool/DbSyncStakePoolProvider/StakePoolAverages.ts

```diff
@@ -0,0 +1,183 @@
+   * This flag is used to let the compute process to know if an async close request
+   * was performed in order to let it interrupt its job.
+   */
+  #interruptComputeProcess = false;
+
+  /**
+   * The logger object.
+   */
+  #logger: Logger;
+
+  /**
+   * The provider used to perform queries on StakePool
+   */
+  #stakePoolProvider: StakePoolProvider;
+
+  /**
+   * The internal status of the compute job
+   */
+  #status: 'computed' | 'error' | 'init' | 'loading' = 'init';
+
+  constructor(dependencies: StakePoolAveragesDependencies) {
+    this.#logger = dependencies.logger;
+    this.#stakePoolProvider = dependencies.stakePoolProvider;
+  }
+
+  async close() {
+    this.#logger.debug('Closing StakePoolAverages provider...');
+
+    // If at least a compute job was started
+    if (this.#cachedResult) {
+      // Instruct the compute process (if it is running) to interrupt its job
+      this.#interruptComputeProcess = true;
+
+      // Wait for the compute process is stopped
+      try {
+        await this.#cachedResult;
+      } catch {
+        // If previous running compute job exited with error, not a problem: we need to exit
+        this.#logger.warn('Running compute job exited with error');
+      }
+    }
+
+    this.#logger.debug('StakePoolAverages provider closed');
+  }
+
+  /**
+   * The compute job
+   */
+  private async compute() {
+    this.#logger.debug('Starting StakePoolAverages compute job...');
+    this.#status = 'loading';
+
+    const PAGE_LENGTH = 50;
+
+    // Init the query options at the first page
+    const queryOptions: StakePoolQueryOptions = {
+      filters: { status: [Cardano.StakePoolStatus.Active] },
+      pagination: { limit: PAGE_LENGTH, startAt: 0 }
+    };
+
+    let lastEpochFound = -1;
+    let moreResults: boolean;
+    let marginPartialSum = 0n;
+    let marginPartialTotal = 0;
+    let rewardPartialSum = 0;
+    let rewardPartialTotal = 0;
+
+    const updatePartialResult = (pool: Cardano.StakePool) => {
+      const { epochRewards, metrics } = pool;
+
+      // No rewards for this pool, skip it
+      if (epochRewards.length === 0) return;
+
+      const lastEpochReward = epochRewards[epochRewards.length - 1];
+
+      // Reward for this pool refers to an old epoch, skip it
+      if (lastEpochReward.epoch < lastEpochFound) return;
+
+      // Reward for this pool refers to an epoch not yet found, reset partials
+      if (lastEpochReward.epoch > lastEpochFound) {
+        lastEpochFound = lastEpochReward.epoch;
+        marginPartialSum = 0n;
+        marginPartialTotal = 0;
+        rewardPartialSum = 0;
+        rewardPartialTotal = 0;
+      }
+
+      marginPartialSum += lastEpochReward.totalRewards;
+      marginPartialTotal++;
+
+      if (metrics.apy !== undefined) {
+        rewardPartialSum += metrics.apy;
+        rewardPartialTotal++;
+      }
+    };
+
+    do {
+      const { pageResults, totalResultCount } = await this.#stakePoolProvider.queryStakePools(queryOptions);
```

Comment:
`queryStakePools` is quite heavy and not fully cached. I suggest

- replace StakePoolProvider dependency with Pool (db) dependency
- write ligher queries for just the data that StakePoolAverages needs

Reply:
> write ligher queries for just the data that StakePoolAverages needs

I'll try to dig into the `queries.ts` file, but I do not exclude I'll need some support on this: I still don't know the schema very well.